cfspopcon/formulas/atomic_data/coeff_interpolator.py

```diff
@@ -0,0 +1,135 @@
+"""Class definition for CoeffInterpolator which is used to interpolate atomic data."""
+
+from functools import partial
+from typing import Union
+
+import numpy as np
+import xarray as xr
+from scipy.interpolate import RectBivariateSpline
+
+from cfspopcon.unit_handling import Quantity, get_units, magnitude_in_units, ureg, wraps_ufunc
+
+
+class CoeffInterpolator(RectBivariateSpline):
+    """An extension of a 2D spline interpolator which handles interpolations in log-space, for interpolating atomic data."""
+
+    tiny = np.finfo(np.float64).tiny
+
+    def log10_with_floor(self, x: Union[xr.DataArray, np.ndarray, float]) -> Union[xr.DataArray, np.ndarray, float]:
```

Comment:
```suggestion
    def log10_with_floor(self, x: xr.DataArray | np.ndarray | float) -> xr.DataArray | np.ndarray | float:
```
Starting with python 3.10, the recommended way to type hint unions of types is through the bitwise-or operator, see: https://docs.python.org/3/library/stdtypes.html#types-union